CI unstable test `TestStoreGlobalConfig` failed

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
in ci https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_coverage/detail/tidb_ghpr_coverage/1228/pipeline
```bash
2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [ERROR] [domain.go:514] ["global config syncer store failed"] [error="context deadline exceeded"] [errorVerbose="context deadline exceeded\ngithub.com/pingcap/errors.AddStack\n\t/nfs/cache/mod/github.com/pingcap/errors@v0.11.5-0.20211009033009-93128226aaa3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/nfs/cache/mod/github.com/pingcap/errors@v0.11.5-0.20211009033009-93128226aaa3/juju_adaptor.go:15\ngithub.com/pingcap/tidb/ddl/util.PutKVToEtcd\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/util/syncer.go:156\ngithub.com/pingcap/tidb/domain/globalconfigsync.(*GlobalConfigSyncer).StoreGlobalConfig\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/domain/globalconfigsync/globalconfig.go:63\ngithub.com/pingcap/tidb/domain.(*Domain).globalConfigSyncerKeeper\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/domain/domain.go:512\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"] [stack="github.com/pingcap/tidb/domain.(*Domain).globalConfigSyncerKeeper\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/domain/domain.go:514"]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [domain.go:506] ["globalConfigSyncerKeeper exited."]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [domain.go:685] ["domain closed"] ["take time"=8.154945532s]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [db.go:567] ["Closing database"]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [db.go:592] ["Memtable flushed"]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [db.go:596] ["Compaction finished"]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [db.go:615] ["BlobManager finished"]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [db.go:619] ["ResourceManager finished"]

[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [INFO] [db.go:625] ["Waiting for closer"]

[2021-11-17T19:17:05.472Z] --- FAIL: TestStoreGlobalConfig (8.56s)

[2021-11-17T19:17:05.472Z]     globalconfig_test.go:79: 

[2021-11-17T19:17:05.472Z]         	Error Trace:	globalconfig_test.go:79

[2021-11-17T19:17:05.472Z]         	Error:      	Not equal: 

[2021-11-17T19:17:05.472Z]         	            	expected: 0

[2021-11-17T19:17:05.472Z]         	            	actual  : 1

[2021-11-17T19:17:05.472Z]         	Test:       	TestStoreGlobalConfig

[2021-11-17T19:17:05.472Z] FAIL

```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

some error msgs
```
[2021-11-17T19:17:05.453Z] [2021/11/18 03:11:21.656 +08:00] [WARN] [syncer.go:153] ["[ddl] etcd-cli put kv failed"] [key=/global/config/enable_resource_metering] [value=true] [error="rpc error: code = Unavailable desc = transport is closing"] [retryCnt=0]
...
[2021-11-17T19:17:05.458Z] [2021/11/18 03:11:23.686 +08:00] [WARN] [syncer.go:153] ["[ddl] etcd-cli put kv failed"] [key=/global/config/enable_resource_metering] [value=true] [error="context deadline exceeded"] [retryCnt=1]
...
[2021-11-17T19:17:05.462Z] [2021/11/18 03:11:25.718 +08:00] [WARN] [syncer.go:153] ["[ddl] etcd-cli put kv failed"] [key=/global/config/enable_resource_metering] [value=true] [error="context deadline exceeded"] [retryCnt=2]
...
[2021-11-17T19:17:05.467Z] [2021/11/18 03:11:27.750 +08:00] [WARN] [syncer.go:153] ["[ddl] etcd-cli put kv failed"] [key=/global/config/enable_resource_metering] [value=true] [error="context deadline exceeded"] [retryCnt=3]
...
[2021-11-17T19:17:05.471Z] [2021/11/18 03:11:29.781 +08:00] [WARN] [syncer.go:153] ["[ddl] etcd-cli put kv failed"] [key=/global/config/enable_resource_metering] [value=true] [error="context deadline exceeded"] [retryCnt=4]
[2021-11-17T19:17:05.472Z] [2021/11/18 03:11:29.812 +08:00] [ERROR] [domain.go:514] ["global config syncer store failed"] [error="context deadline exceeded"] [errorVerbose="context deadline exceeded\ngithub.com/pingcap/errors.AddStack\n\t/nfs/cache/mod/github.com/pingcap/errors@v0.11.5-0.20211009033009-93128226aaa3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/nfs/cache/mod/github.com/pingcap/errors@v0.11.5-0.20211009033009-93128226aaa3/juju_adaptor.go:15\ngithub.com/pingcap/tidb/ddl/util.PutKVToEtcd\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/util/syncer.go:156\ngithub.com/pingcap/tidb/domain/globalconfigsync.(*GlobalConfigSyncer).StoreGlobalConfig\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/domain/globalconfigsync/globalconfig.go:63\ngithub.com/pingcap/tidb/domain.(*Domain).globalConfigSyncerKeeper\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/domain/domain.go:512\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"] [stack="github.com/pingcap/tidb/domain.(*Domain).globalConfigSyncerKeeper\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/domain/domain.go:514"]

```

--- a/domain/globalconfigsync/globalconfig_test.go
+++ b/domain/globalconfigsync/globalconfig_test.go
@@ -73,10 +73,22 @@ func TestStoreGlobalConfig(t *testing.T) {
 	_, err = se.Execute(context.Background(), "set @@global.tidb_enable_top_sql=1;")
 	require.NoError(t, err)
 
-	resp, err := cluster.RandClient().Get(context.Background(), "/global/config/enable_resource_metering")
-	require.NoError(t, err)
-	require.NotNil(t, resp, nil)
-	require.Equal(t, len(resp.Kvs), 1)
-	require.Equal(t, resp.Kvs[0].Key, []byte("/global/config/enable_resource_metering"))
-	require.Equal(t, resp.Kvs[0].Value, []byte("true"))
+	for i := 0; i < 20; i++ {
+		resp, err := cluster.RandClient().Get(context.Background(), "/global/config/enable_resource_metering")
+		require.NoError(t, err)
+		require.NotNil(t, resp, nil)
+
+		if len(resp.Kvs) == 0 {
+			// writing to ectd is async, so we should retry if not synced yet
+			time.Sleep(100 * time.Millisecond)
+			continue
+		}
+
+		require.Equal(t, len(resp.Kvs), 1)
+		require.Equal(t, resp.Kvs[0].Key, []byte("/global/config/enable_resource_metering"))
+		require.Equal(t, resp.Kvs[0].Value, []byte("true"))
+		return
+	}
+
+	require.Fail(t, "timeout for waiting etcd synced")
 }

